File: python/KRAKEN/Tests/DW/Manual/kupert.py
#==================================================================
#  
#  KRAKEN: Top roughness manual case
#  Mexilhoeira Grande, Sab Jul 29 10:28:28 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run kupert

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("KRAKEN - Top roughness manual case")

system("kraken.exe kupert")
system("fields.exe kupert < kupert.flp")

filename = 'kupert.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure )
tl = -20*log10( abs( p ) )
tlmin = min( tl )
tlmax = max( tl )

figure(1)
plot(rarraykm,tl)
xlabel('Range (km)')
ylabel('TL (dB)')
title('KRAKEN - Top roughness manual case')
ylim(tlmax,tlmin)
grid(True)

show()

print("done.")

File: python/SCOOTER/Tests/DW/Manual/kupert.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('SCOOTER - Top roughness manual case')

system('cp fields.flp kupert.flp');
system('scooter.exe kupert');
system('fields.exe kupert < kupert.flp');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'kupert.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

figure(1)
plot(rarraykm,tl)
xlabel('Range (km)')
ylabel('TL (dB)')
title('SCOOTER - Top roughness manual case')
grid on, box on
axis ij

disp('done.')
